### ISSUE TITLE
Cannot add Jupyter notebook extension

### ISSUE THREAD

**Issue by nathancarter** [gap-packages/francy#33]
On Mac OS X, I followed the instructions in the README file, with the following results.

```
$ pip install jupyter_francy
```

Got errors about insufficient permissions, because I set up my pip as root for some reason, so...

```
$ sudo pip install jupyter_francy
```

Did the whole installation, completing with this line of output:

```
Successfully installed jupyter-francy-0.8.10b0
```

Then this step failed, as shown:

```
$ jupyter nbextension enable --py --sys-prefix jupyter_francy
Traceback (most recent call last):
  File "/usr/local/Cellar/jupyter/1.0.0_1/libexec/bin/jupyter-nbextension", line 11, in <module>
    load_entry_point('notebook==5.0.0', 'console_scripts', 'jupyter-nbextension')()
  File "/usr/local/Cellar/jupyter/1.0.0_1/libexec/lib/python2.7/site-packages/jupyter_core/application.py", line 267, in launch_instance
    return super(JupyterApp, cls).launch_instance(argv=argv, **kwargs)
  File "/usr/local/Cellar/jupyter/1.0.0_1/libexec/vendor/lib/python2.7/site-packages/traitlets/config/application.py", line 658, in launch_instance
    app.start()
  File "/usr/local/Cellar/jupyter/1.0.0_1/libexec/lib/python2.7/site-packages/notebook/nbextensions.py", line 900, in start
    super(NBExtensionApp, self).start()
  File "/usr/local/Cellar/jupyter/1.0.0_1/libexec/lib/python2.7/site-packages/jupyter_core/application.py", line 256, in start
    self.subapp.start()
  File "/usr/local/Cellar/jupyter/1.0.0_1/libexec/lib/python2.7/site-packages/notebook/nbextensions.py", line 808, in start
    self.toggle_nbextension_python(self.extra_args[0])
  File "/usr/local/Cellar/jupyter/1.0.0_1/libexec/lib/python2.7/site-packages/notebook/nbextensions.py", line 784, in toggle_nbextension_python
    logger=self.log)
  File "/usr/local/Cellar/jupyter/1.0.0_1/libexec/lib/python2.7/site-packages/notebook/nbextensions.py", line 445, in enable_nbextension_python
    logger=logger)
  File "/usr/local/Cellar/jupyter/1.0.0_1/libexec/lib/python2.7/site-packages/notebook/nbextensions.py", line 360, in _set_nbextension_state_python
    m, nbexts = _get_nbextension_metadata(module)
  File "/usr/local/Cellar/jupyter/1.0.0_1/libexec/lib/python2.7/site-packages/notebook/nbextensions.py", line 1034, in _get_nbextension_metadata
    m = import_item(module)
  File "/usr/local/Cellar/jupyter/1.0.0_1/libexec/vendor/lib/python2.7/site-packages/traitlets/utils/importstring.py", line 42, in import_item
    return __import__(parts[0])
ImportError: No module named jupyter_francy
```

Because `pip list` does not mention `jupyter_francy`, but rather `jupyter-francy`, I also tried

```
$ jupyter nbextension enable --py --sys-prefix jupyter-francy
```

but that produced the same results (except with `-` in place of `_` of course).

What am I doing wrong?

**Comment by mcmartins** [comment]
Hi @nathancarter, humm something happened after release v0.5.6... Will investigate and come back to you ASAP...

**Comment by nathancarter** [comment]
There seems to be an additional installation step [here](https://jupyter-notebook.readthedocs.io/en/stable/extending/frontend_extensions.html#installing-and-enabling-extensions) before enabling nbextensions now.  I tried doing that for `jupyter_francy`, but couldn't seem to get the right combination of commands together.  Just mentioning this as a potential place to start hunting for the problem.

**Comment by nathancarter** [comment]
Update:

The following procedure seems to install correctly:

```
$ jupyter nbextension install /path/to/where/pip/puts/your/python/stuff/jupyter_francy/
$ jupyter nbextension enable jupyter_francy/nbextension/extension --sys-prefix
```

But then when I run the notebook, I get script errors in `require` calls from francy.  I tried it with `index` instead of `extension`, but with the same result.

**Comment by mcmartins** [comment]
@nathancarter, thanks for the update. I messed up with something at some point... Since I'm working on a new feature to allow multiple renderers and did alot of changes, I've decided to upload the latest version into pypi. 

I've just tried to install everything from scratch and it works fine. Make sure you remove all the pip francy modules installed, remove the extension from jupyter and then run:

```
(test-extension) mcmartins@local:~$ pip install jupyter_francy
(test-extension) mcmartins@local:~$ jupyter nbextension enable --py --sys-prefix jupyter_francy
```

Will install version 0.9.0. 

It's in pretty early stage but I'll release a final version in the next days.

**Comment by nathancarter** [comment]
Thanks for helping out with this!  Unfortunately, I still have some problems.

I think I have successfully removed the old jupyter_francy, as you can see here:
```
$ jupyter nbextension list
Known nbextensions:
  config dir: /Users/nathan/.jupyter/nbconfig
    notebook section
      gap-mode/main  enabled
      - Validating: OK
      test-extension/main  enabled
      - Validating: OK
```

But when I attempt to install the new one, this happens:
```
$ sudo pip install jupyter_francy
[many bits of console output omitted here]
Installing collected packages: jupyter-francy
Successfully installed jupyter-francy-0.9.0a0
$ jupyter nbextension enable --py --sys-prefix jupyter_francy
Traceback (most recent call last):
  File "/usr/local/Cellar/jupyter/1.0.0_5/libexec/bin/jupyter-nbextension", line 11, in <module>
    load_entry_point('notebook==5.5.0', 'console_scripts', 'jupyter-nbextension')()
  File "/usr/local/Cellar/jupyter/1.0.0_5/libexec/lib/python3.7/site-packages/jupyter_core/application.py", line 266, in launch_instance
    return super(JupyterApp, cls).launch_instance(argv=argv, **kwargs)
  File "/usr/local/Cellar/jupyter/1.0.0_5/libexec/vendor/lib/python3.7/site-packages/traitlets/config/application.py", line 658, in launch_instance
    app.start()
  File "/usr/local/Cellar/jupyter/1.0.0_5/libexec/lib/python3.7/site-packages/notebook/nbextensions.py", line 988, in start
    super(NBExtensionApp, self).start()
  File "/usr/local/Cellar/jupyter/1.0.0_5/libexec/lib/python3.7/site-packages/jupyter_core/application.py", line 255, in start
    self.subapp.start()
  File "/usr/local/Cellar/jupyter/1.0.0_5/libexec/lib/python3.7/site-packages/notebook/nbextensions.py", line 896, in start
    self.toggle_nbextension_python(self.extra_args[0])
  File "/usr/local/Cellar/jupyter/1.0.0_5/libexec/lib/python3.7/site-packages/notebook/nbextensions.py", line 872, in toggle_nbextension_python
    logger=self.log)
  File "/usr/local/Cellar/jupyter/1.0.0_5/libexec/lib/python3.7/site-packages/notebook/nbextensions.py", line 483, in enable_nbextension_python
    logger=logger)
  File "/usr/local/Cellar/jupyter/1.0.0_5/libexec/lib/python3.7/site-packages/notebook/nbextensions.py", line 380, in _set_nbextension_state_python
    m, nbexts = _get_nbextension_metadata(module)
  File "/usr/local/Cellar/jupyter/1.0.0_5/libexec/lib/python3.7/site-packages/notebook/nbextensions.py", line 1122, in _get_nbextension_metadata
    m = import_item(module)
  File "/usr/local/Cellar/jupyter/1.0.0_5/libexec/vendor/lib/python3.7/site-packages/traitlets/utils/importstring.py", line 42, in import_item
    return __import__(parts[0])
ModuleNotFoundError: No module named 'jupyter_francy'
```

Do we need the separate install step mentioned above, or am I barking up the wrong tree?

Thank you again.

**Comment by mcmartins** [comment]
Hi @nathancarter , no worries. I really appreciate that you're trying the module out :)
That is strange, jupyter cannot find the module so I'm wondering if you have multiple versions of python installed? Perhaps pip is pointing to python2? Can you please check if you have a pip3? Jupyter requires python 3 and indeed I can see it in the logs _(.../libexec/lib/python3.7/...)_, but could the **jupyter_francy** extension been installed on the python2 environment instead?

**Comment by nathancarter** [comment]
Ah, that's it!  My apologies for my lack of knowledge of my own configuration.  *Replacing pip with pip3 made it possible to install jupyter_francy after that*, and I can confirm that launching jupyter with a GAP kernel correctly loads the jupyter_francy module (based on what the console says).  I haven't tried to use it yet, but this step is all set!  Thank you very much.

**Comment by mcmartins** [comment]
:D no problem @nathancarter! I believe I have introduced some bugs in the version you installed :P, so in the next days I'll release a fully working version and advise you to use it.
Once again, thanks for trying this out.

**Comment by mcmartins** [comment]
Released version v0.10.0
